### PR TITLE
Revised `general-superstaq` docstrings

### PR DESCRIPTION
**Pull request by bharat-thotakura**
Part of the 'Update old docstrings' bullet point of https://github.com/Infleqtion/client-superstaq/issues/563 in gss.

### CODE REVIEW

File: general-superstaq/general_superstaq/superstaq_client.py [hunk 1]
```diff
@@ -213,22 +218,27 @@ def _accept_terms_of_use(self, user_input: str) -> str:
...
 
         Returns:
             String with success message.
         """
         return self.post_request("/accept_terms_of_use", {"user_input": user_input})
 
     def get_targets(self) -> Dict[str, Dict[str, List[str]]]:
-        """Makes a GET request to Superstaq API to get a list of available targets."""
+        """Makes a GET request to the Superstaq API to get a list of available, unavailable,
+        and retired targets.
```
Comment (dowusu-antwi) [inline general-superstaq/general_superstaq/superstaq_client.py:230]:
```suggestion
        """Makes a GET request to the Superstaq API.
        
        Gets a list of available, unavailable, and retired targets.
```
(or something like that)

File: general-superstaq/general_superstaq/superstaq_client.py [hunk 2]
```diff
@@ -327,13 +372,14 @@ def supercheq(
         return self.post_request("/supercheq", json_dict)
 
     def target_info(self, target: str) -> Dict[str, Any]:
-        """Makes a POST request to the Superstaq API (using the `/target_info` endpoint to request
-        information about `target`.
+        """Makes a POST request to the Superstaq API (using the `/target_info` endpoint to
+        request information about `target`.
```
Comment (dowusu-antwi) [inline general-superstaq/general_superstaq/superstaq_client.py:376]: Make 1-liner (unsure how best to split it up)

**Review by dowusu-antwi — APPROVED** [review]
A couple of style comments, o/w LGTM.

File: general-superstaq/general_superstaq/superstaq_client.py (hunk 1 above)
Comment (bharat-thotakura) [reply general-superstaq/general_superstaq/superstaq_client.py]: Done in [f98da3c](https://github.com/Infleqtion/client-superstaq/pull/590/commits/f98da3c10266d3819e15851de8e7e4c78304640f)

File: general-superstaq/general_superstaq/superstaq_client.py (hunk 2 above)
Comment (bharat-thotakura) [reply general-superstaq/general_superstaq/superstaq_client.py]: Done in [f98da3c](https://github.com/Infleqtion/client-superstaq/pull/590/commits/f98da3c10266d3819e15851de8e7e4c78304640f)

**Pull request merged**